packages/flutter/lib/src/widgets/_html_element_view_io.dart

```diff
@@ -0,0 +1,34 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: prefer_const_constructors_in_immutables
+// ignore_for_file: avoid_unused_constructor_parameters
+
+import 'framework.dart';
+import 'platform_view.dart';
+
+/// The platform-specific implementation of [HtmlElementView].
+extension HtmlElementViewImpl on HtmlElementView {
+  /// Creates an [HtmlElementView] that renders a DOM element with the given
+  /// [tagName].
+  static HtmlElementView createFromTagName({
+    Key? key,
+    required String tagName,
+    bool isVisible = true,
+    ElementCreatedCallback? onElementCreated,
+  }) {
+    return HtmlElementView(key: key, viewType: 'never');
```

Comment:
I made it this way to avoid an issue we had before: https://github.com/flutter/flutter/issues/43532

But now that I think about it, I think that issue doesn't apply to this case because this is a factory (not `const`).

Reply:
I would never expect anybody to undonditionally call this factory (or the HtmlElementView constructor) from non-web code. I can't think of the use case for that. Maybe do:

```
if (kIsWeb)
  myWebWidget,
if (!kIsWeb)
  myIoWidget,
```

in a `<Widget>[]` or something? It's fine if you want to leave it as this, but I'm curious about the breaking code :/

Reply:
If you do this:
```dart
if (kIsWeb) myWebWidget
```
with a `const` constructor, it'll be constructed at "compile time" and throws (before it checks for the `kIsWeb` condition). That was the original problem.

But as I said, this particular code (`HtmlElementView.fromTagName()`) is not a `const` constructor, so the problem doesn't exist. (it is still a problem for the main `HtmlElementView()` constructor, so we can't do an assert there).

Reply:
Ahhh yes, and our lints push hard for all constructors to be `const` 👍